src/easystar.js

```diff
@@ -533,9 +533,9 @@ EasyStar.js = function() {
             var dx = Math.abs(x1 - x2);
             var dy = Math.abs(y1 - y2);
             if (dx < dy) {
-                return DIAGONAL_COST * dx + dy;
+                return DIAGONAL_COST * dx + (dy-dx);
             } else {
-                return DIAGONAL_COST * dy + dx;
+                return DIAGONAL_COST * dy + (dx=dy);
```

Comment:
@mikeparker sorry i know its old, i implemented this change myself also but wanted to ask... why was it dx=dy

Reply:
Yeah it's pretty old, that looks like a typo to me, it's probably supposed to be `(dx-dy)`

If I was better with javascript this is exactly the sort of thing a unit test should check for.

Reply:
Yes that makes sense thinking about it. If dx < dy then you travel all the X distance via diagonals then the remainer dy via vertical moves.

If its the other way around then you travel all the Y distance via diagonals then the remainder dx via horizontal moves.

Reply:
I've fixed the typo.